tracing-tracy/src/lib.rs

```diff
@@ -144,6 +163,10 @@ impl Default for TracyLayer {
     }
 }
 
+thread_local! {
+    static CACHE: VecCell<String> = const { VecCell::new() };
```

Comment:
FWIW I had great success using `crossbeam_queue::ArrayQueue` for this sort of thing in the past. It may be somewhat slower on the account that it is a global queue (and thus there is going to cache lane contention and whatnot.) Benefit is that you no longer need to worry about situations where a span is created in one thread, moved into another and dropped, or worry about `!Send`/`!Sync` issues, as the `ArrayQueue` is both.

And if `ArrayQueue` turns out to be a poor fit for this use-case, then this `CACHE` definitely needs at least an upper bound on how many elements it can store here.

Reply:
That's a pretty clever idea! However, I'm fairly opposed to adding a central contention point for all threads. Given that Tracy claims to be a nanosecond resolution tracer, I'm pretty sure we'd be losing that claim with a shared cache line. Or at the very least, I would want to see benchmarks of trying to write to an atomic from 16 threads simultaneously to see what the latency looks like.